broker-util/oo-admin-repair

```diff
@@ -403,6 +418,37 @@ def find_unresponsive_servers(available_servers, auto_confirm)
   unresponsive_servers
 end
 
+def find_orphaned_envs(namespace=nil)
+  orphaned_envs = []
+  domains = []
+  if namespace
+    begin
+      domains = [Domain.find_by(:namespace => namespace)]
+    rescue Mongoid::Errors::DocumentNotFound
+      print_message "Unable to find domain. The domain #{namespace} does not exist. Skipping..."
+    rescue Exception => e
+      print_message "Unable to find domain #{namespace} with error: #{e.message}."
+    end
+    # Find all component_ids for every application in this domain
+    component_ids = domains[0].applications.map {|a| a.component_instances.map {|ci| ci._id}}.flatten
+  else
+    domains = Domain.all
+    # find all component_ids for every application
+    component_ids = Application.all.map {|a| a.component_instances.map {|ci| ci._id}}.flatten
```

Comment:
I'm pretty sure I can reduce the load on the database as much as possible by using `Application.only(:component_instances).pluck(:_id)`, which should return *only* the array we need from the database. I'm not sure if we will still need a warning when the query is going to be "too large" to take a reasonable amount of time.

@a13m could you provide any input on this? Would the previously mentioned query be too large of a query to perform within a reasonable time on very large environments like Online? If so, do you have a guesstimate of how many applications would be considered "too large" of a query?